### PR TITLE
Added `set_records` and `set_record` to update sheet via records.[Issue #677]

### PR DESCRIPTION
**Pull request by muddi900**

### CODE REVIEW

**Comment by alifeee** [comment]
thanks for the implementation!

I find the test hard to parse, would it be possible to write it with very explicit objects? e.g.,

```python
table_before = [
  ["header 1", "header 2"]
  ["value 1",  "value 2"]
]
table_after = [
  ["header 1",    "header 2"]
  ["value 1",     "value 2"]
  ["new value 1", "new value 2"]
]
```

File: gspread/worksheet.py [hunk 1]
```diff
@@ -606,6 +618,150 @@ def get_all_records(
...
+                raise GSpreadException("Extra headers found in the data set")
+
+            insert_row = []
+            for col in cols:
+                insert_row.append(row.get(col, default_blank))
+            insert_rows.append(insert_row)
+
+        self.append_rows(
+            insert_rows,
+            value_input_option=ValueInputOption.user_entered,
```
Comment (lavigne958) [inline gspread/worksheet.py:654]: why should we force the user to use that valueInputOption type ? :thinking:

File: gspread/worksheet.py [hunk 2]
```diff
@@ -606,6 +618,150 @@ def get_all_records(
...
+
+            ins_row = []
+            for col in cols:
+                ins_row.append(row.get(col, default_blank))
+            insert_rows.append(ins_row)
+
+        self.insert_rows(
+            insert_rows,
+            row=insert_row,
+            value_input_option=ValueInputOption.user_entered,
```
Comment (lavigne958) [inline gspread/worksheet.py:726]: same here

File: gspread/worksheet.py [hunk 3]
```diff
@@ -197,6 +197,17 @@ def __init__(
...
+
+    @property
+    def column_headers(self) -> List[str]:
+        if not self._column_headers:
+            self._column_headers = self.row_values(1)
+        return self._column_headers
+
+    @column_headers.setter
+    def column_headers(self, value: List[str]) -> None:
+        self._column_headers = value
```
[the remark this answers is not in the document]
Comment (muddi900) [reply gspread/worksheet.py]: Fixed this as well
